readthedocs/notifications/notification.py

```diff
@@ -39,9 +37,9 @@ class Notification:
     send_email = True
     extra_tags = ''
 
-    def __init__(self, context_object, request, user=None):
+    def __init__(self, context_object, request=None, user=None):
         self.object = context_object
-        self.request = request
+        self.request = request or HttpRequest()
```

Comment:
we were requiring passing this parameter, and always using `HttpRequest` when we didn't have it, so looks like a good default. Doesn't look like we are using it for making decisions.